src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -205,7 +205,11 @@ export class TxSender {
         }
         this.onComplete(undefined, confirmCallback)
       })
-      .then(({ transactionHash }) => transactionHash)
+      .then((receipt) => {
+        if (didTxRevert(receipt)) {
```

Comment:
Interesting, I always thought it throws in this case.
Do we need the same fix for the Safe creation tx?

Reply:
I did as well, but it seemingly doesn't.

We should definitely add it to the Safe creation transaction as well. I'll create a branch from this as it doesn't relate to pending statuses.

Reply:
Here's the PR: https://github.com/gnosis/safe-react/pull/3692

Reply:
Seems like it may be throwing:

![image](https://user-images.githubusercontent.com/20442784/158812789-fc1c741a-51f2-4dca-b37f-e3ca5711f22f.png)

I'm moving both of these back into in progress.

Reply:
In this instance it will throw as it is tied to the `promiEvent`. If we retrieve the receipt manually though, it would return the receipt without throwing.